noxfile.py

```diff
@@ -0,0 +1,127 @@
+        session.run("pybamm_install_jax")
+    session.run("coverage", "run", "--rcfile=.coveragerc", "run-tests.py", "--nosub")
+    session.run("coverage", "combine")
+    session.run("coverage", "xml")
+
+
+@nox.session(name="integration", reuse_venv=True)
+def run_integration(session):
+    homedir = os.getenv("HOME")
+    session.env["SUNDIALS_INST"] = session.env.get("SUNDIALS_INST", f"{homedir}/.local")
+    session.env[
+        "LD_LIBRARY_PATH"
+    ] = f"{homedir}/.local/lib:{session.env.get('LD_LIBRARY_PATH')}"
+    session.run("pip", "install", "-e", ".")
+    if sys.platform == "linux":
+        session.install("scikits.odes")
+    session.run("python", "run-tests.py", "--integration")
+
+
+@nox.session(name="doctests", reuse_venv=True)
+def run_doctests(session):
+    session.run("pip", "install", "-e", ".[docs]")
+    session.run("python", "run-tests.py", "--doctest")
+
+
+@nox.session(name="unit", reuse_venv=True)
+def run_unit(session):
+    homedir = os.getenv("HOME")
+    session.env["SUNDIALS_INST"] = session.env.get("SUNDIALS_INST", f"{homedir}/.local")
+    session.env[
+        "LD_LIBRARY_PATH"
+    ] = f"{homedir}/.local/lib:{session.env.get('LD_LIBRARY_PATH')}"
+    session.run("pip", "install", "-e", ".")
+    if sys.platform == "linux":
+        session.run("pybamm_install_jax")
+        session.install("scikits.odes")
+    session.run("python", "run-tests.py", "--unit")
+
+
+@nox.session(name="examples", reuse_venv=True)
+def run_examples(session):
+    session.run("pip", "install", "-e", ".[dev]")
+    session.run("python", "run-tests.py", "--examples")
+
+
+@nox.session(name="dev", reuse_venv=True)
+def set_dev(session):
+    homedir = os.getenv("HOME")
+    LD_LIBRARY_PATH = f"{homedir}/.local/lib:{session.env.get('LD_LIBRARY_PATH')}"
+    envbindir = session.bin
+    session.run("pip", "install", "-e", ".[dev]")
+    session.install("cmake")
+    session.run(
+        "echo",
+        "export",
+        f"LD_LIBRARY_PATH={LD_LIBRARY_PATH}",
+        ">>",
+        f"{envbindir}/activate",
+    )
+
+
+@nox.session(name="tests", reuse_venv=True)
+def run_tests(session):
+    homedir = os.getenv("HOME")
+    session.env["SUNDIALS_INST"] = session.env.get("SUNDIALS_INST", f"{homedir}/.local")
+    session.env[
+        "LD_LIBRARY_PATH"
+    ] = f"{homedir}/.local/lib:{session.env.get('LD_LIBRARY_PATH')}"
+    session.run("pip", "install", "-e", ".")
+    if sys.platform == "linux":
+        session.run("pybamm_install_jax")
+        session.install("scikits.odes")
+    session.run("python", "run-tests.py", "--all")
+
+
+@nox.session(name="docs", reuse_venv=True)
+def build_docs(session):
+    envbindir = session.bin
+    session.run("pip", "install", "-e", ".")
+    session.install(
+        "sphinx>=1.5",
+        "pydata-sphinx-theme",
+        "sphinx-autobuild",
+        "sphinx_design",
+        "sphinx-copybutton",
+        "myst-parser",
+        "sphinx-inline-tabs",
+    )
```

Comment:
I think this should work?
```suggestion
    session.run("pip", "install", "-e", ".[docs]")
```